openflow-sdn/openflow-protocol.md

```diff
@@ -0,0 +1,65 @@
+
+Before moving further, the term "switch" denotes a network device capable of using OF protocol, and not the Layer 2 device. 
+
+## Background on traditional SDN
+
+**Software Defined Networking** is a novel concept for the network engineering field which enables traditional networks to be controlled through a separate centralized entity, usually an SDN controller. In simple terms, **SDN is defined as separating the _control plane_(or the brain) of the network and the _data/forwarding plane_(or the brawn)**.
+
+The SDN controller is a centralized physical/virtual device that communicates with all the "dumb" network devices and updates them on how to forward traffic. This southbound communication takes place through means like an SSH connection, APIs, and most commonly through the OpenFlow(OF) protocol. This protocol defines the requirements and the standards for communication between an SDN controller and the "agent" network device.
+
+In traditional SDN architecture, there are three layers:
+- Application layer: This is where the applications will run and decide how the traffic should move in a network. For example, routing protocols like OSPF, BGP, Firewall application, etc. 
+- Control Layer: This is where the controller resides. The controller acts as a bridge between the application and the switches. It relays the information from the application and converts them into network "flows" that are entered in the switch's "flow table". 
+- Data/ Infrastructure layer: This is the forwarding plane that has the actual hardware devices. These devices refer to the flow entries in their flow table to forward packets through the network. 
+
+## Basics of Openflow
+
+As mentioned, the OpenFlow protocol lays out the foundation for communication between an SDN controller and a dumb network device. This protocol was developed first by researchers at Stanford University in 2008 and was first adopted by [Google](https://static.googleusercontent.com/media/research.google.com/en//pubs/archive/42948.pdf) in their backbone network in 2011-2012. It is managed now by the [Open Networking Foundation(ONF)](https://opennetworking.org/sdn-definition/). The latest version used in the industry is V1.5. 
+
+OpenFlow is the standard southbound protocol used between the SDN controller and the switch. The SDN controller takes the information from the applications and converts them into flow entries which are fed to the switch via OF. It can also be used for monitoring switch and port statistics for network management.
+
+**NOTE**
+The OpenFlow protocol is only established between a controller and the switch. It does not affect the rest of the network and if a packet capture were to be taken between two switches in a network, both of them connected to the controller via another port, the packet capture would not reveal any OF messages between the switches. It is strictly for the use between a switch and the controller. The rest of the network is not affected.
+
```

Comment:
```suggestion
OpenFlow is the standard southbound protocol used between the SDN controller and the switch. The SDN controller takes the information from the applications and converts them into flow entries, which are fed to the switch via OF. It can also be used for monitoring switch and port statistics for network management.

**NOTE**
The OpenFlow protocol is only established between a controller and the switch. It does not affect the rest of the network. If a packet capture were to be taken between two switches in a network, both connected to the controller via another port, the packet capture would not reveal any OF messages between the switches. It is strictly for the use between a switch and the controller. The rest of the network is not affected.
```